### PR TITLE
feat(docker): Add image layers handler

### PR DESCRIPTION
**Pull request by lucashalbert**
This PR facilitates generating architecture variant layer badges such as `arm/v6` VS `arm/v7` VS `arm64/v8`. It fulfills the layer feature requests in issue [#185](https://github.com/badgen/badgen.net/issues/185). Unfortunately, the DockerHub Registry API Scheme v2 does not support `blob` layers, and instead shows `fsLayers` which do not include layers of size 0 (ie: `CMD`, `ENV`, `LABEL`, `ENTRYPOINT`, etc...). For this reason, users will notice a difference in the number of layers between microbadger and badgen.net.

This can be tested using the multi-architecture [lucashalbert/curl:latest](https://hub.docker.com/r/lucashalbert/curl)  dockerhub image like so:

* [https://badgen.net/docker/layers/lucashalbert/curl/latest/arm/v6](https://badgen.net/docker/layers/lucashalbert/curl/latest/arm/v6).
* [https://badgen.net/docker/layers/lucashalbert/curl/latest/arm/v7](https://badgen.net/docker/layers/lucashalbert/curl/latest/arm/v7).
* [https://badgen.net/docker/layers/lucashalbert/curl/latest/arm64/v8](https://badgen.net/docker/layers/lucashalbert/curl/latest/arm64/v8).

### CODE REVIEW

**Review by amio — APPROVED** [review]
Great work! @lucashalbert 

> the DockerHub Registry API Scheme v2 does not support blob layers, and instead shows fsLayers which do not include layers of size 0 (ie: CMD, ENV, LABEL, ENTRYPOINT, etc...). For this reason, users will notice a difference in the number of layers between microbadger and badgen.net.

We may put this note on the [docker](https://badgen.net/docker) page [with `help` field](https://github.com/badgen/badgen.net/blob/master/api/uptime-robot.ts#L5-L18), just like [Uptime Robot](https://badgen.net/uptime-robot).

**Comment by lucashalbert** [comment]
@amio - After exploring the DockerHub Registry API a bit more, I found where all layers (including those of size 0) are stored for particular images. Can we hold off on merging this PR until I have a chance to add this functionality?

**Comment by lucashalbert** [comment]
@amio, my new commit [7a90fd3](https://github.com/badgen/badgen.net/pull/442/commits/7a90fd30d3d6b183bd3a7630931d9f540c702b12) refactors the codebase to query metadata (size, layers [including layers of size 0], and others defined in [Label Schema Convention](http://label-schema.org/)) directly from the image config hosted on the DockerHub registry. This allows the Badgen.net Docker API to be highly extensible by directly querying any metadata associated with an image.

Here are the examples that I built out.
metadata (size) image [https://badgen.net/docker/metadata/size/lucashalbert/curl/latest/arm/v6](https://badgen.net/docker/metadata/size/lucashalbert/curl/latest/arm/v6)
metadata (layers and size) image [https://badgen.net/docker/metadata/layers/lucashalbert/curl/latest/arm/v7](https://badgen.net/docker/metadata/layers/lucashalbert/curl/latest/arm/v7)
metadata (layers) [icon & label] image [https://badgen.net//docker/metadata/layers/lucashalbert/curl/latest/arm/v7?icon=docker&label=layers](https://badgen.net//docker/metadata/layers/lucashalbert/curl/latest/arm/v7?icon=docker&label=layers)
metadata (layers) [label] image [https://badgen.net//docker/metadata/layers/lucashalbert/curl/latest/arm/v7?label=docker%20layers](https://badgen.net//docker/metadata/layers/lucashalbert/curl/latest/arm/v7?label=docker%20layers)
metadata (version) image [https://badgen.net//docker/metadata/version/lucashalbert/curl/latest/arm64/v8](https://badgen.net//docker/metadata/version/lucashalbert/curl/latest/arm64/v8)
metadata (architecture) image [https://badgen.net//docker/metadata/architecture/lucashalbert/curl/latest/arm64/v8](https://badgen.net//docker/metadata/architecture/lucashalbert/curl/latest/arm64/v8)
metadata (build-date) image https://badgen.net//docker/metadata/build-date/lucashalbert/curl/latest/arm64/v8
metadata (maintainer) image https://badgen.net//docker/metadata/maintainer/lucashalbert/curl/latest/arm64/v8

**Review by amio — COMMENTED** [review]
Awesome!

Can we drop `/metadata` in the path?

**Comment by lucashalbert** [comment]
@amio, `/metadata` has been removed from the 'layers' path.

I do, however, believe that the ability to query Label-Schema format metadata from an image is extremely useful. For that reason, I've built out the metadata handler.

File: api/docker.ts
Comment (amio) [inline api/docker.ts:230]: Let's keep the convention that subject/label is a static descriptive text

File: api/docker.ts
Comment (amio) [inline api/docker.ts:151]: We can put these into `libs/docker.ts`, throw a `BadgenError` for unknown image, like [github badge](https://github.com/badgen/badgen.net/blob/370a4a87b6c3278965923528b7156b0250446561/libs/github.ts#L32).

File: api/docker.ts
Comment (lucashalbert) [reply api/docker.ts]: This was done in commit [254fef7](https://github.com/badgen/badgen.net/pull/442/commits/254fef7691ab207ae87c188d1406d8082534b1a6)

File: api/docker.ts
Comment (lucashalbert) [reply api/docker.ts]: I moved all of the token and error checking logic into libs\docker.ts with commit [1d62a11](https://github.com/badgen/badgen.net/pull/442/commits/1d62a11926ccca73f8fab8246d7c799c51702dd2)

**Review by amio — APPROVED** [review]
Thanks @lucashalbert , great work!

**Pull request merged**
